app/jobs/petition_mobile_sync_worker.rb

```diff
@@ -0,0 +1,21 @@
+class PetitionMobileSyncWorker
+  include Shoryuken::Worker
+  shoryuken_options queue: Rails.application.secrets.queues['petition_mobile_sync'], auto_delete: true
+
+  attr_reader :repository
+  attr_reader :mobile_api_service
+
+  def initialize(repository: PetitionPlugin::DetailVersionRepository.new, mobile_api_service: MobileApiService.new)
+    @repository = repository
+    @mobile_api_service = mobile_api_service
+  end
+
+  def perform(sqs_msg, body)
+    petition_detail_version_id = JSON.parse(body)['id']
+
+    Rails.logger.info "Synchronizing petition version: #{petition_detail_version_id}"
+
+    version = repository.find_by_id! petition_detail_version_id
```

Comment:
This queue will retry indefinitely if the record is not found. You log and mark this as run normally.